api/repositories/build_repository_test.go

```diff
@@ -595,12 +613,104 @@ var _ = Describe("BuildRepository", func() {
 				createRoleBinding(ctx, userName, spaceDeveloperRole.Name, namespace1.Name)
 			})
 			It("if user is a Space Developer it returns a list with one element", func() {
-				buildRecords, fetchError = buildRepo.ListBuilds(ctx, authInfo)
+				buildRecords, fetchError = buildRepo.ListBuilds(ctx, authInfo, listMessage)
 				Expect(fetchError).NotTo(HaveOccurred())
 				Expect(buildRecords).To(HaveLen(1))
 				Expect(buildRecords[0].GUID).To(Equal(build1GUID))
 			})
 		})
+		When("the app_guids filter is provided", func() {
```

Comment:
I would suggest moving all the `When("XXX filter is provided"` under `When("the user is a space developer"`. Thus you would not have to create the role binding for every filtering `When`